packages/amplify-cli/src/extensions/amplify-helpers/resource-status.js

```diff
@@ -8,8 +8,9 @@ const { getEnvInfo } = require('./get-env-info');
 const _ = require('lodash');
 const { CLOUD_INITIALIZED, CLOUD_NOT_INITIALIZED, getCloudInitStatus } = require('./get-cloud-init-status');
 const { readJsonFile } = require('./read-json-file');
+const { ServiceName: FunctionServiceName, hashLayerResource } = require('amplify-category-function');
 
-async function isBackendDirModifiedSinceLastPush(resourceName, category, lastPushTimeStamp) {
+async function isBackendDirModifiedSinceLastPush(resourceName, category, lastPushTimeStamp, isLambdaLayer = false) {
```

Comment:
It is not generic enough, as we'll have to fix hashing for other services (AppSync), I suggest to pass in the service name here and not a boolean.

Reply:
agreed this needs a refactor, but this was implemented to close a gap that was discovered late last week and there wasn't time for a larger overhaul